src/apps/proj_strtod.cpp

```diff
@@ -123,6 +123,13 @@ double proj_strtod(const char *str, char **endptr) {
         return 0;
     }
 
+    /* NaN */
+    if (strncmp(p, "NaN", 3) == 0) {
+        if (endptr)
+            *endptr = const_cast<char *>(p + 3);
+        return NAN;
+    }
```

Comment:
I'm not sure how strict/flexible this code needs to be. It specifically checks for only NaN (not "nan", not "NAN", etc). It also doesn't do anything if the character after NaN is a whitespace. For a valid NaN it should be, but right now someone could technically enter the next coordinate number right after a NaN (ex. `NaN0.0m` would be two values I think).